router/core/websocket.go

```diff
@@ -121,19 +121,31 @@ func NewWebsocketMiddleware(ctx context.Context, opts WebsocketMiddlewareOptions
 		handler.forwardQueryParamsConfig.withStaticAllowList = len(handler.forwardQueryParamsConfig.staticAllowList) > 0
 		handler.forwardQueryParamsConfig.withRegexAllowList = len(handler.forwardQueryParamsConfig.regexAllowList) > 0
 	}
-	if opts.EnableWebSocketEpollKqueue {
-		poller, err := epoller.NewPoller(opts.EpollKqueueConnBufferSize, opts.EpollKqueuePollTimeout)
-		if err == nil {
-			opts.Logger.Debug("Epoll is available")
-
-			handler.epoll = poller
-			handler.connections = make(map[int]*WebSocketConnectionHandler)
-			go handler.runPoller()
+	if opts.EnableNetPoll {
+		if err := netpoll.Supported(); err != nil {
```

Comment:
Is there a way we can simplify this nested if else? It's not so easy to read IMO

Reply:
Probably, but with more efforts. A single nested if should be fine.